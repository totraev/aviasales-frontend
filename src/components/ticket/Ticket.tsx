import React, { SFC } from "react";
import styles from "./Ticket.module.css";

import BuyButton from "../../containers/buyButton/BuyButton";
import Seporator from "./Seporator";

import { formatDate } from "../../helpers/date";

import { ITicketProps } from "./Ticket.types";

const Ticket: SFC<ITicketProps> = ({ ticket }) => (
  <div className={styles.ticket}>
    <div className={styles.carrier}>
      <img
        className={styles.logo}
        srcSet={`
          ${require("./images/logo.png")},
          ${require("./images/logo@2x.png")} 2x,
          ${require("./images/logo@3x.png")}
        `}
        src={require("./images/logo@3x.png")}
        alt="TK"
      />

      <BuyButton price={ticket.price} />
    </div>

    <div className={styles.info}>
      <div className={styles.segment}>
        <span className={styles.time}>{ticket.departure_time}</span>
        <Seporator stops={ticket.stops} />
        <span className={styles.time}>{ticket.arrival_time}</span>
      </div>

      <div className={styles.segment}>
        <span className={styles.origin}>
          {ticket.origin}, {ticket.origin_name}
        </span>
        <span className={styles.destination}>
          {ticket.destination_name}, {ticket.destination}
        </span>
      </div>

      <div className={styles.segment}>
        <span className={styles.date}>{formatDate(ticket.departure_date)}</span>
        <span className={styles.date}>{formatDate(ticket.arrival_date)}</span>
      </div>
    </div>
  </div>
);

export default Ticket;
