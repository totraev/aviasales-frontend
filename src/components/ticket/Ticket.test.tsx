import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import Ticket from "./Ticket";

describe("Ticket component", () => {
  it("Should renders correctly", () => {
    const ticket = {
      arrival_date: "12.05.18",
      arrival_time: "23:50",
      carrier: "S7",
      departure_date: "12.05.18",
      departure_time: "17:20",
      destination: "TLV",
      destination_name: "Тель-Авив",
      origin: "VVO",
      origin_name: "Владивосток",
      price: 13100,
      stops: 1
    };
    const renderer = createRenderer();
    const tree = renderer.render(<Ticket ticket={ticket} />);

    expect(tree).toMatchSnapshot();
  });
});
