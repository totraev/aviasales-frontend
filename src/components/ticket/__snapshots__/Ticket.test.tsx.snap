// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket component Should renders correctly 1`] = `
<div
  className="ticket"
>
  <div
    className="carrier"
  >
    <img
      alt="TK"
      className="logo"
      src="logo@3x.png"
      srcSet="
          logo.png,
          logo@2x.png 2x,
          logo@3x.png
        "
    />
    <Connect(BuyButton)
      price={13100}
    />
  </div>
  <div
    className="info"
  >
    <div
      className="segment"
    >
      <span
        className="time"
      >
        17:20
      </span>
      <Seporator
        stops={1}
      />
      <span
        className="time"
      >
        23:50
      </span>
    </div>
    <div
      className="segment"
    >
      <span
        className="origin"
      >
        VVO
        , 
        Владивосток
      </span>
      <span
        className="destination"
      >
        Тель-Авив
        , 
        TLV
      </span>
    </div>
    <div
      className="segment"
    >
      <span
        className="date"
      >
        5 дек. 2018, ср
      </span>
      <span
        className="date"
      >
        5 дек. 2018, ср
      </span>
    </div>
  </div>
</div>
`;
